### PR TITLE
DAOS-4818 dtx: Pass DTX to QUERY and ENUM

### PR DESCRIPTION
**Pull request by liw**
As a follow-on change to #2776, this patch addresses QUERY_KEY and ENUM
operations. It also makes these cleanups:

  - Remove parameter th from daos_obj_query.
  - Replace shard_query_key_args.kqa_epoch with
    shard_query_key_args.kqa_auxi.epoch.
  - Remove the unused obj_query_key_cb_args.flags.

Signed-off-by: Li Wei <wei.g.li@intel.com>

### CODE REVIEW

**Comment by liw** [comment]
The REBUILD failures are DAOS-4943.

**Comment by liw** [comment]
Merely rebased to the latest master; no changes.

**Comment by liw** [comment]
The test failures are not related to this PR:

- rebuild "25": dtx issue fixed already in the latest master
- The rest: DAOS-4943 and DAOS-4471

**Comment by liw** [comment]
@Nasf-Fan, any thoughts on this PR, please?

**Comment by johannlombardi** [comment]
@jolivier23 @Nasf-Fan could you please prioritize this review?

File: src/object/obj_rpc.h [hunk 1]
```diff
@@ -254,13 +255,16 @@ CRT_RPC_DECLARE(obj_key_enum, DAOS_ISEQ_OBJ_KEY_ENUM, DAOS_OSEQ_OBJ_KEY_ENUM)
...
 
 #define DAOS_ISEQ_OBJ_QUERY_KEY	/* input fields */		 \
+	((struct dtx_id)	(okqi_dti)		CRT_VAR) \
 	((uuid_t)		(okqi_co_hdl)		CRT_VAR) \
 	((uuid_t)		(okqi_pool_uuid)	CRT_VAR) \
 	((uuid_t)		(okqi_co_uuid)		CRT_VAR) \
 	((daos_unit_oid_t)	(okqi_oid)		CRT_VAR) \
 	((uint64_t)		(okqi_epoch)		CRT_VAR) \
 	((uint32_t)		(okqi_map_ver)		CRT_VAR) \
+	((uint32_t)		(okqi_api_flags)	CRT_VAR) \
```
Comment (Nasf-Fan) [inline src/object/obj_rpc.h:265]: I would suggest to use 64-bits for 'api_flags' to be consistent with others, such as orw_api_flags, and opi_api_flags.

Comment (liw) [reply src/object/obj_rpc.h:265]: Agreed. Will refresh.

**Comment by liw** [comment]
Functional cart-self-test: CART-900

**Pull request merged**